sys/linux/auto.txt

```diff
@@ -181,19 +308,91 @@ getdents64$auto(fd fd, dirent ptr[inout, linux_dirent64$auto], count int32)
 getgroups$auto(gidsetsize int32, grouplist ptr[inout, int32])
 getgroups32$auto(gidsetsize int32, grouplist ptr[inout, int32])
 getitimer$auto(which int32, value ptr[inout, __kernel_old_itimerval$auto])
+getitimer$auto_ITIMER_PROF(which const[ITIMER_PROF], value ptr[inout, __kernel_old_itimerval$auto])
+getitimer$auto_ITIMER_REAL(which const[ITIMER_REAL], value ptr[inout, __kernel_old_itimerval$auto])
+getitimer$auto_ITIMER_VIRTUAL(which const[ITIMER_VIRTUAL], value ptr[inout, __kernel_old_itimerval$auto])
 getpeername$auto(fd fd, usockaddr ptr[inout, sockaddr], usockaddr_len ptr[inout, int32])
 getpgid$auto(pid pid) pid
 getpriority$auto(which int32, who uid)
+getpriority$auto_PRIO_PGRP(which const[PRIO_PGRP], who uid)
+getpriority$auto_PRIO_PROCESS(which const[PRIO_PROCESS], who uid)
+getpriority$auto_PRIO_USER(which const[PRIO_USER], who uid)
 getrandom$auto(ubuf ptr[inout, string], len intptr, flags int32)
 getresgid$auto(rgidp ptr[inout, int32], egidp ptr[inout, int32], sgidp ptr[inout, int32])
 getresgid32$auto(rgidp ptr[inout, int32], egidp ptr[inout, int32], sgidp ptr[inout, int32])
 getresuid$auto(ruidp ptr[inout, int32], euidp ptr[inout, int32], suidp ptr[inout, int32])
 getresuid32$auto(ruidp ptr[inout, int32], euidp ptr[inout, int32], suidp ptr[inout, int32])
 getrlimit$auto(_resource int32, rlim ptr[inout, rlimit$auto])
 getrusage$auto(who int32, ru ptr[inout, rusage$auto])
+getrusage$auto_RUSAGE_BOTH(who const[RUSAGE_BOTH], ru ptr[inout, rusage$auto])
+getrusage$auto_RUSAGE_CHILDREN(who const[RUSAGE_CHILDREN], ru ptr[inout, rusage$auto])
+getrusage$auto_RUSAGE_SELF(who const[RUSAGE_SELF], ru ptr[inout, rusage$auto])
 getsid$auto(pid pid) pid
 getsockname$auto(fd fd, usockaddr ptr[inout, sockaddr], usockaddr_len ptr[inout, int32])
 getsockopt$auto(fd fd, level int32, optname int32, optval ptr[inout, string], optlen ptr[inout, int32]) fd
+getsockopt$auto_SO_ACCEPTCONN(fd fd, level int32, optname const[SO_ACCEPTCONN], optval ptr[inout, string], optlen ptr[inout, int32]) fd
```

Comment:
There are many cases when the syscall variants only differ in the single const valued argument.

Do you think we will be able to extract more details about their arguments in the near future so that they are actually different? If not, it would be nice to create an enum in such cases and have just one syscall variant.

Otherwise, we might really want to start considering the improvements to our `O(N^2)` call priority algorithm in `prog` and the way we auto-generate `sys/*/gen/*.go` files.

Reply:
I would gave up on the idea that we have few syscalls, and not spend time and complicate code trying to remove a dozen now and then.
We may remove few dozens of syscalls now, just to add several hundreds tomorrow for another interface, and then hundreds more day after tomorrow.
So unless we can remove half of syscalls, I wouldn't bother, it won't solve O(N^2) problem for long.

Reply:
I will look into removing auto-generated Go files.
Is call priority algorithm a problem? Do we know that it takes long?

Reply:
> Is call priority algorithm a problem? Do we know that it takes long?

It's not a huge problem since it it's updated in background, but
1) It will be taking increasingly more CPU time to update
2) It will be taking increasingly more space. How many "syscalls" do we have now? 5000? That's minimum 100 MB just to represent it.
3) I'm not sure its efficiency stays the same.